src/shared/json-queue.c

```diff
@@ -124,29 +123,36 @@ cJSON * jqueue_parse_json(file_queue * queue) {
         if (end = strchr(buffer, '\n'), end) {
             *end = '\0';
 
-            if ((object = cJSON_ParseWithOpts(buffer, &jsonErrPtr, 0), object) && (*jsonErrPtr == '\0')) {
-                queue->read_attempts = 0;
+            if ((object = cJSON_ParseWithOpts(buffer, &jsonErrPtr, 0), object) && (*jsonErrPtr == '\0')) {   
                 return object;
             }
-        }
-
-        // The read JSON is invalid
-        if (object) {
+            
+            // The read JSON is invalid                                   
             cJSON_Delete(object);
+
+            mwarn("Invalid JSON alert read from '%s': '%s'", queue->file_name, buffer);
+            return NULL;
         }
 
-        queue->read_attempts++;
-        mdebug2("Invalid JSON alert read from '%s'. Remaining attempts: %d", queue->file_name, MAX_READ_ATTEMPTS - queue->read_attempts);
+        while (strlen(buffer) == OS_MAXSTR)
+        {            
+            if(fgets(buffer, OS_MAXSTR + 1, queue->fp)){
 
-        if (queue->read_attempts < MAX_READ_ATTEMPTS) {
-            if (current_pos >= 0) {
-                if (fseek(queue->fp, current_pos, SEEK_SET) != 0) {
-                    queue->flags = CRALERT_READ_FAILED;
+                if (strchr(buffer, '\n')){
+                    mwarn("Overlong JSON alert read from '%s'", queue->file_name);
+                    return NULL;
                 }
+            } else {
+                break;
```

Comment:
If the `fgets` fails, you should set the flag `CRALERT_READ_FAILED` and return `NULL` to avoid reading this again.

Reply:
If `fgets` fails, we should read the entire alert again, to avoid incorrectly warning of an `invalid JSON alert`, when the function is called the next time.

`clearerr` was added before the second jqueue_parse_json call.

I think adding a `sleep` call is not necessary, because one of them already exists in integratord.c while loop, just after `jqueue_next` call.